lib/dry/system/errors.rb

```diff
@@ -92,5 +92,36 @@ def initialize(plugin, message, gem = nil)
         super("dry-system plugin #{plugin.inspect} failed to load its dependencies: #{details}")
       end
     end
+
+    # Exception raised when auto-registerable component is not loadable
+    #
+    # @api public
+    ComponentNotLoadableError = Class.new(NameError) do
+      # @api private
+      def initialize(component, error, corrections: DidYouMean::ClassNameChecker.new(error).corrections)
+        full_class_name = [error.receiver, error.name].join("::")
+
+        message = [
+          "Component '#{component.key}' with auto_register: true is not loadable.",
```

Comment:
```suggestion
          "Component '#{component.key}' is not loadable.",
```

Just noticed this (sorry!), and consistent with the thoughts I left before, I reckon we could even do away with the mention of "auto_register: true" inside the error message, since I think this will be implicitly clear to our users, and in most cases, they won't be explicitly mentioning `auto_register: true` on the components, so this detail might be more confusing than not.

Reply:
Ah yes, of course!